### PR TITLE
Support `where` argument in ufuncs

### LINKED ISSUE #7332: Bitwise operators mask (where parameter)
It seems that dask does not implement the `where` parameter in the bitwise operations but numpy does. It would be great to implement those to increase compatibility of the current function.

Here is the documentation in number for invert (bitwise_not)
https://numpy.org/doc/stable/reference/generated/numpy.invert.html#numpy.invert

An example in dask below would produce an error. Substituting numpy with dask would work.

```
import dask.array as da
x = da.zeros((10)).astype(np.uint8)
mask = da.ones((10)).astype(bool)
x = da.bitwise_not(x, where=mask)
```

### LINKED ISSUE #8285: handle_out() returns nothing if out is set
**What happened**: function set o to the output dask array and returned None.

**What you expected to happen**: function set o to the output dask array and return the output array.

**Minimal Complete Verifiable Example**:

```py
>>> from dask import array as da
>>> x = da.array( [1,2,3] )
>>>
>>> print( da.invert( x ) )
dask.array<invert, shape=(3,), dtype=int64, chunksize=(3,), chunktype=numpy.ndarray>
>>>
>>> o = da.array( [4,5,6] )
>>> print( da.invert( x, out=o ) )
None
```

**Anything else we need to know?**:

**Environment**:

- Dask version: dask-2021.9.1+38.geab6a998
- Python version: 3.9.7
- Operating System: Manjaro
- Install method (conda, pip, source): source

### PR DESCRIPTION
**Pull request by ek234**
Closes #7332 
Fixes #8285
Supersedes #8322

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by quasiben** [comment]
add to allowlist

**Comment by ncclementi** [comment]
@ek234 I see the CI hasn't run all the tests because there are conflicts. I believe that these conflicts are causing some checks not passing, notice that one says "AppVeyor was unable to build non-mergeable pull request" and the gpuCI is probably failing for the same reason. 
You might want to merge main into your branch, fix the conflicts, and push again. That way the full CI suite will run and if we get any fails we will be able to assist you better.

**Comment by ek234** [comment]
Thanks for the review
I did as you said and then went through the readthedocs errors

Let me know if you would like me to make any changes.

**Comment by ek234** [comment]
bruh it's all green now:P

**Comment by ncclementi** [comment]
@ek234 thank you for your patience, I'm pinging another maintainer to see if they can review the code. 

@jrbourbeau Would you mind reviewing this PR? We are at a point that we need a maintainer to take a look. I've suggested some changes, but I think we need a deeper look.

**Comment by jcrist** [comment]
Thanks for all your work here @ek234, and apologies for the delayed review. This issue turned out to be a bit more complicated than a good-first-issue (apologies for the mislabeling). Since `where` interacts with `out` (which mutates the passed array), and dask arrays are effectively immutable, this required some nuanced changes to how `elemwise` works internally. I've pushed up a commit to your branch that makes the required changes on top of your initial PR. Thanks for your work here, hope to get this merged soon.

**Comment by douglasdavis** [comment]
Looks like this PR may take care of what https://github.com/dask/dask/pull/8322 is trying to accomplish

**Comment by jcrist** [comment]
Ah, that it does. I've added it to the top-level comment to auto-close.

**Comment by ian-r-rose** [comment]
@jcrist Is there anything left to be done here? Looks like CI is now passing, though there are now merge conflicts to be resolved.

File: dask/array/core.py [hunk 1]
```diff
@@ -4367,7 +4367,7 @@ def broadcast_shapes(*shapes):
     return tuple(reversed(out))
 
 
-def elemwise(op, *args, **kwargs):
+def elemwise(op, *args, out=None, where=True, dtype=None, name=None, **kwargs):
     """Apply elementwise function across arguments
 
     Respects broadcasting rules
```
Comment (ncclementi) [inline dask/array/core.py:4373]: Can we update these docstrings and examples to show how `where` works?

Comment (jcrist) [reply dask/array/core.py:4373]: I've updated the docstring. `where` matches the numpy behavior and is rarely used, figure it's not worth adding a contrived example.

**Comment by jcrist** [comment]
Thanks for the ping, I'd forgotten about this. Merge conflicts fixed, should be good to merge once tests pass.

File: dask/array/core.py (hunk 1 above)
Comment (ian-r-rose) [reply dask/array/core.py]: I agree it's rarely used, but I do find it useful surprisingly often for avoiding divide-by-zero issues

**Comment by jcrist** [comment]
Thanks @ek234, this is in.

**Pull request merged**